graph/State-Associated-Subsets.md

```diff
@@ -0,0 +1,113 @@
+
+## Solution
+--------
+
+Have an abstract base class that all 'variants' of the subset are then derived types from the base subset. You can find general subtyping guidance [here](https://github.com/microsoft/api-guidelines/blob/op-graphPatterns/graph/Modelling%20with%20Subtypes%20Pattern.md).
+
+The abstract base class should also hold an enum for all possible variants. The purpose of including this is to allow for easier ways to do query and filter operations on variants like 'all' and 'none' without relying on isof functions.
+
+Base Type
+```xml
+    <ComplexType Name="membership" IsAbstract="true">
+      <Property Name="membershipKind" Type="graph.membershipKind"/>
+    </ComplexType>
+
+    <Enum Name="membershipKind">
+      <Member Name="all"/>
+      <Member Name="enumerated"/>
+      <Member Name="none"/>
+      <Member Name="unknownFutureValue"/>
+    </Enum>
+```
+
+Derived Types
+```xml
+    <ComplexType Name="noMembership" BaseType="graph.membership"/>
+
+    <ComplexType Name="allMembership" BaseType="graph.membership"/>
+
+    <ComplexType Name="enumeratedMembership" BaseType = "graph.membership">
+      <Property Name="members" Type="Collection(Edm.String)"/>
+    </ComplexType>
+
+    <ComplexType Name="excludedMembership" BaseType = "graph.membership">
+      <Property Name="members" Type="Collection(Edm.String)"/>
+    </ComplexType>
+```
+
+## Issues and Considerations
+-------------------------
+
+Given we are using overarching subtype model, subtyping model limitations apply here as well - see [subtyping documentation](https://github.com/microsoft/api-guidelines/blob/op-graphPatterns/graph/Modelling%20with%20Subtypes%20Pattern.md) for more details.
+ 
+
+## When to Use this Pattern
+------------------------
+
+This pattern is to be used when supporting two or more collection states of the following where at least one of the states is a subset variant.
+- All targets
+- No targets
+- Subset of targets to be included
+- Subset of targets to be excluded.
+
+If you only **ever** need to support two states - All or None - without usage of any subsets, it would be better to use a Boolean to toggle on and off.
+
+## Example
+-------
+
+GET https://graph.microsoft.com/v1.0/identity/conditionalAccess/policies/
+
+_Note: unrelated properties on entities omitted for easier readability_
+
+```json
+{
+    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#conditionalAccessPolicy",
+    "values": [
+        {
+            "id": "66d36273-fe4c-d478-dc22-e0179d856ce7",
+            "conditions": {
+                "users": {
+                    "includeGuestsOrExternalUsers": {
+                        "externalTenants": {
+                            "@odata.type":"microsoft.graph.conditionalAccessAllExternalTenants",
```

Comment:
I am struggling to remember from the discussion Mike led her, but wht is the value of having BOTH the odata.type annotation AND the membershipKind enum?   Isn't the odata.typ eon its own adequate?  Perhaps I am just forgetting the reason.